server/src/server.js

```diff
@@ -778,15 +778,15 @@ app.post("/api/delete-shot", function(req, res) {
 });
 
 app.post("/api/disconnect-device", function(req, res) {
-  if (!req.deviceId) {
+  if (!req.deviceId && !req.accountId) {
     sendRavenMessage(req, "Attempt to disconnect without login");
     simpleResponse(res, "Not logged in", 401);
     return;
   }
   disconnectDevice(req.deviceId).then((result) => {
     const keygrip = dbschema.getKeygrip("auth");
     const cookies = new Cookies(req, res, {keys: keygrip});
-    if (result) {
+    if (result || req.accountId) {
```

Comment:
We could remove this if statement entirely and simply always clear the account cookie. If the statement isn't true then no response is sent, so the logic wasn't right to start with.